tests/TestGIN.py

```diff
@@ -1,104 +1,154 @@
+        for i in range(len(causal_order)):
+            assert np.isclose(causal_order[i], ground_truth[i]).all()
 
+    def test_case1_hsic(self):
+        sample_size = 500
+        random.seed(42)
+        np.random.seed(42)
+        L1 = np.random.uniform(-1, 1, size=sample_size)
+        L2 = np.random.uniform(1.2, 1.8) * L1 + np.random.uniform(-1, 1, size=sample_size)
+        X1 = np.random.uniform(1.2, 1.8) * L1 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X2 = np.random.uniform(1.2, 1.8) * L1 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X3 = np.random.uniform(1.2, 1.8) * L2 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X4 = np.random.uniform(1.2, 1.8) * L2 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
         data = np.array([X1, X2, X3, X4]).T
         data = (data - np.mean(data, axis=0)) / np.std(data, axis=0)
-        g, k = GIN(data)
-        print(g, k)
-
-        # Visualization using pydot
-        from causallearn.utils.GraphUtils import GraphUtils
-        pyd = GraphUtils.to_pydot(g)
-        tmp_png = pyd.create_png(f="png")
-        fp = io.BytesIO(tmp_png)
-        img = mpimg.imread(fp, format='png')
-        plt.axis('off')
-        plt.imshow(img)
-        plt.show()
-
-    def test_case2(self):
-        sample_size = 1000
-        np.random.seed(0)
-        L1 = np.random.uniform(-1, 1, size=sample_size) ** 5
-        L2 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        L3 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1,
-                                                                                                     size=sample_size) ** 5
-        X1 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X2 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X3 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X4 = np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X5 = np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X6 = np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X7 = np.random.uniform(0.5, 2.0) * L3 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X8 = np.random.uniform(0.5, 2.0) * L3 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X9 = np.random.uniform(0.5, 2.0) * L3 + np.random.uniform(-1, 1, size=sample_size) ** 5
+        _, causal_order = GIN(data, indep_test_method='hsic')
+        causal_order = [sorted(cluster_i) for cluster_i in causal_order]
+        ground_truth = [[0, 1], [2, 3]]
+        assert len(causal_order) == len(ground_truth)
+        for i in range(len(causal_order)):
+            assert np.isclose(causal_order[i], ground_truth[i]).all()
 
+    def test_case2_kci(self):
```

Comment:
same for test case 2.

Basically you can test both kci and hsic in a single function, and the only difference seems to be GIN(data, indep_test_method) indep_test_method this parameter. 

Check https://github.com/cmu-phil/causal-learn/pull/59